src/test/java/htsjdk/samtools/SamReaderFactoryTest.java

```diff
@@ -290,4 +308,98 @@ public void inputResourceFromStringTest() throws IOException {
       Assert.assertEquals(SamInputResource.of("/a/b/c").data().type(),
           InputResource.Type.FILE);
     }
+
+    @Test
+    public void testCRAMReaderFromURL() throws IOException {
+        getCRAMReaderFromInputResource(
+                (cramURL, indexURL) -> { return SamInputResource.of(cramURL).index(indexURL);},
+                true,
+                3);
+    }
+
+    @Test
+    public void testCRAMReaderFromURLStream() throws IOException {
+        getCRAMReaderFromInputResource(
+                (cramURL, indexURL) -> {
+                    try {
+                        ISeekableStreamFactory streamFactory = SeekableStreamFactory.getInstance();
+                        return SamInputResource
+                                .of(streamFactory.getStreamFor(cramURL))
+                                .index(streamFactory.getStreamFor(indexURL));
+                    }
+                    catch (IOException e) {
+                        throw new RuntimeIOException(e);
+                    }
+                },
+                true,
+                3);
+    }
+
+    @Test
+    public void testCRAMReaderFromURLNoIndexFile() throws IOException {
+        getCRAMReaderFromInputResource(
+                (cramURL, indexURL) -> { return SamInputResource.of(cramURL); },
+            false,
+            11);
+    }
+
+    @Test(expectedExceptions=RuntimeIOException.class)
+    public void testCRAMReaderFromURLBadIndexFile() throws IOException {
+        getCRAMReaderFromInputResource(
+                (cramURL, indexURL) -> { return SamInputResource.of(cramURL).index(new File("nonexistent.bai")); },
+            true,
```

Comment:
This test case is slightly confusing -- recommend adding a one-line comment for each test case